sxg_rs/src/header_integrity.rs

```diff
@@ -106,6 +107,15 @@ impl<'a, F: Fetcher, C: HttpCache> HeaderIntegrityFetcher for HeaderIntegrityFet
 }
 
 impl<'a, F: Fetcher, C: HttpCache> HeaderIntegrityFetcherImpl<'a, F, C> {
+    async fn cache_get(&self, url: &str) -> Result<HttpResponse> {
+        self.header_integrity_cache.try_borrow_mut()?.get(url).await
+    }
+    async fn cache_put(&self, url: &str, response: &HttpResponse) -> Result<()> {
+        self.header_integrity_cache
+            .try_borrow_mut()?
```

Comment:
`RefCell::try_borrow_mut` does not wait the resource to be available, hence it will skip writing to cache if `header_integrity_cache` is being borrowed by others. This may reduce the cache hit rate. We can change `HttpCache` from `RefCell` to `Mutex` (such as [tokio](https://docs.rs/tokio/1.11.0/tokio/sync/struct.Mutex.html) or [async-std](https://docs.rs/async-std/1.10.0/async_std/sync/struct.Mutex.html)) in a future PR.

Reply:
Thanks, good catch. I was able to write a test to demonstrate this cache miss, so I'll work on it in this PR. My first attempt at solving it using async_std::Mutex resulted in a deadlock, so I'm going to take a break from this PR for a day.

Reply:
OK, PTAL. I solved this by adding more interior mutability. JsHttpCache doesn't require mut. InMemoryCache does, and RefCell is good enough for test.